deps/rabbitmq_stream/src/rabbit_stream_reader.erl

```diff
@@ -1747,7 +1747,11 @@ handle_frame_post_auth(Transport,
                                internal_sequence = InternalSequence + 1},
                              State}
                     end;
-                {_, _} ->
+                {PublisherIdTaken, ReferenceTaken} ->
+                    rabbit_log:warning("Error while declaring publisher for stream '~tp', "
+                                       "with reference '~tp': ID already taken ~tp, "
+                                       "reference already taken ~tp",
```

Comment:
Can we make these say "ID already taken: …, reference already taken: …", or perhaps "ID already taken? …, reference already taken? …"

And I agree with Gabriele that we probably should log the ID. Not sure if the reference would be useful to the user or to ourselves. But we can log both for completeness' sake.

Reply:
I pushed a change to add the publisher ID:

```
2024-02-09 09:49:09.789164+01:00 [warning] <0.804.0> Error while declaring publisher 2 for stream '<<"declarePublisherUniqueness-15237b97-6ef2-4323-ab6b-91b950d36b79-s1">>', with reference '<<"ref-1">>'. ID already taken: false. Reference already taken: true.
```

What do you think?

Reply:
LGTM

Reply:
Wait, surely we can format those binaries as "regular strings"?